match/lib/match/nuke.rb

```diff
@@ -55,6 +55,7 @@ def run(params, type: nil)
         s3_secret_access_key: params[:s3_secret_access_key].to_s,
         s3_bucket: params[:s3_bucket].to_s,
         s3_object_prefix: params[:s3_object_prefix].to_s,
+        s3_skip_encryption: params[:s3_skip_encryption],
         gitlab_project: params[:gitlab_project],
         team_id: params[:team_id] || Spaceship::ConnectAPI.client.portal_team_id
       })
```

Comment:
I kinda dislike how the current design of `for_storage_mode` forces all these unrelated params to be littered across all the call sites... what do you think about refactoring it so we just pass in the whole `params` object plus any overrides and nil replacements? Then this entire block of code could be replace with something like

```ruby
self.storage = Storage
  .from_params(params)
  .replace_if_nil({team_id: Spaceship::ConnectAPI.client.portal_team_id})
```

Reply:
Fine with me, I can give it a go. It was kinda more changes than I thought would be needed since I had to hand pick all params.

Stay tuned :)

Reply:
Okay, so the `params` object in this case is a `FastlaneCore::Configuration` and that does not play nicely with the existing tests... I will see if it something I can work around.

Reply:
@getaaron I gave it a swing, found that it made the most sense to cherry pick the params going into the Storage classes themselves since they are more specific.

Please give me your thoughts.